### PR TITLE
[test][db migrator] Enhancement: handle the case that no buffer change in the latest database version

### PR DESCRIPTION
**Pull request by stephenxs**
#### What I did
Enhancement: handle the case that no buffer change in the latest database version
Current, the following two versions are the same:
- The latest version changed by mellanox_buffer_migrator
- The latest version in CONFIG_DB

That won't be true if another part in CONFIG_DB is updated. In that case, the latest version in CONFIG_DB will be greater than the latest version in mellanox_buffer_migrator.
However, this can break the buffer migrator unit test:
- The db_migrator will always migrate the database to the latest version
- The config database version check will fail in case the latest version in the config database doesn't match that defined in the buffer migrator.

This is to support this case.

Signed-off-by: Stephen Sun <stephens@nvidia.com>

#### How I did it
Advance the database version in the expected config database before version checking if the following conditions satisfied:
- The version in the expected database is the latest version in the Mellanox buffer migrator
- The version in the result database is greater than the latest version in the Mellanox buffer migrator

#### How to verify it
Run unit test.

### CODE REVIEW

**Review by liat-grozovik — APPROVED** [review]

**Comment by liat-grozovik** [comment]
@stephenxs is this fix needed for 202012? if so, please check if it can be cherry picked or need to have new PR against 202012.

**Comment by stephenxs** [comment]
> @stephenxs is this fix needed for 202012? if so, please check if it can be cherry picked or need to have new PR against 202012.

Hi @liat-grozovik 
I see a PR depending on this one (1473) was tagged as "Request for 202012". So, yes, we need it for 202012.
Just checked, it can be cherry-picked smoothly.

**Comment by liat-grozovik** [comment]
@neethajohn could you please review?

File: tests/db_migrator_test.py [hunk 1]
```diff
@@ -173,6 +186,7 @@ def test_mellanox_buffer_migrator_for_warm_reboot(self, sku, topo):
         self.mellanox_buffer_migrator_warm_reboot_runner(input_db_name, input_db_name, expected_db_name, expected_db_name, True)
 
     def test_mellanox_buffer_migrator_negative_nondefault_for_warm_reboot(self):
+        device_info.get_sonic_version_info = get_sonic_version_info_mlnx
```
[the remark this answers is not in the document]
Comment (stephenxs) [reply tests/db_migrator_test.py:189]: Hi @neethajohn 
During db migrator initialization, the `device_info.get_sonic_version_info` is called to get the necessary info.
We need to mock it if we run it for unit test.
FYI, the snipped of code
```
        version_info = device_info.get_sonic_version_info()
        asic_type = version_info.get('asic_type')
        self.asic_type = asic_type
```
The backtrace
```
-> self.mellanox_buffer_migrator_warm_reboot_runner(input_config_db, input_appl_db, expected_config_db, expected_appl_db, False)
  /sonic/src/sonic-utilities/tests/db_migrator_test.py(159)mellanox_buffer_migrator_warm_reboot_runner()
-> dbmgtr = db_migrator.DBMigrator(None)
  /sonic/src/sonic-utilities/scripts/db_migrator.py(71)__init__()
-> version_info = device_info.get_sonic_version_info()
> /sonic/src/sonic-utilities/tests/db_migrator_test.py(25)get_sonic_version_info_mlnx()
-> return {'asic_type': 'mellanox'}
```

**Review by neethajohn — APPROVED** [review]

**Pull request merged**